src/server/lock_manager/deadlock.rs

```diff
@@ -99,7 +99,10 @@ impl DetectTable {
     /// Returns the key hash which causes deadlock.
     pub fn detect(&mut self, txn_ts: TimeStamp, lock_ts: TimeStamp, lock_hash: u64) -> Option<u64> {
         let _timer = DETECTOR_HISTOGRAM_VEC.detect.start_coarse_timer();
```

Comment:
How about the DETECTOR_HISTOGRAM_VEC? (Maybe you can change it in future PR)

Reply:
@breeswish DETECTOR_HISTOGRAM_VEC start a timer at the function call,end at the structure lifetime,I think it is already optimal now.

Reply:
@breeswish Do you mean start_coarse_timer() could be a thread local HistogramTimer?Since HistogramTimer call observe automatically at `drop` method,I think the improvement,if possible,should be  part of `rust-prometheus`,not part of tikv